### PR TITLE
Check for duplicate login in TOSERVER_INIT handler

### PR DESCRIPTION
**Pull request by EliasFleckenstein03**
This PR resolves issue #10957 by checking for a player already connected to the server with the same name in the TOSERVER_INIT handler. You might notice that I did not remove the old code; this is intended, because when to players connect with the same name at the same time the old handler in StageTwoClientInit will kick the one that comes second. This fix does not cause any problems with older clients (tested).

## To do

This PR is Ready for Review.

## How to test

Host a local server and wait until you are connected. Then, try to connect to it with the same name - it will fail before loading media etc.

### CODE REVIEW

**Review by SmallJoker — APPROVED** [review]
Works

File: src/network/serverpackethandler.cpp [hunk 1]
```diff
@@ -180,7 +180,7 @@ void Server::handleCommand_Init(NetworkPacket* pkt)
 	if (player && player->getPeerId() != PEER_ID_INEXISTENT) {
 		actionstream << "Server: Player with name \"" << playername <<
 			"\" tried to connect, but player with same name is already connected" << std::endl;
-			DenyAccess(peer_id, SERVER_ACCESSDENIED_ALREADY_CONNECTED);
```
Comment (EliasFleckenstein03) [inline src/network/serverpackethandler.cpp]: I don't fully understand why you changed this. The previous indentation seems right to me. The `DenyAccess(...)` statement is a child of the `if`, and not the `actionstream << ...`.

**Comment by sfan5** [comment]
> because when to players connect with the same name at the same time the old handler in StageTwoClientInit will kick the one that comes second

I don't think this duplication is possible anyway but we can just leave both checks in.

**Review by sfan5 — APPROVED** [review]

**Comment by EliasFleckenstein03** [comment]
I tested it, the duplication is possible.

**Comment by nerzhul** [comment]
it's strange that duplicate is possible. For me before we kicked the previous session.
What strategy ? forbid or kick previous one ?

**Pull request merged**